include/envoy/http/header_map.h

```diff
@@ -467,6 +476,17 @@ class HeaderMap {
    */
   virtual void setReferenceKey(const LowerCaseString& key, const std::string& value) PURE;
 
+  /**
+   * Replaces a header value by copying the value. Copies the key if the key does not exist.
+   *
+   * Calling setCopy multiple times for the same header will result in only the last header
+   * being present in the HeaderMap.
+   *
+   * @param key specifies the name of the header to set; it WILL be copied.
+   * @param value specifies the value of the header to set; it WILL be copied.
+   */
+  virtual void setCopy(const LowerCaseString& key, const std::string& value) PURE;
```

Comment:
As long as you are in here should we just update all of these to string_view per @lizan other comment?